Add basic account filter/search

bturner-r7 opened the pull request:
This adds a very rudimentary search function for recent logins.  It's a simple case insenstive search.  It doesn't re-order results or anything fancy like that.  This should make it easier to quickly find accounts if you have given them helpful aliases.

Currently it searches both the name and url.  Searching the URL may not make sense, but it didn't hurt in my tests.

<img width="796" alt="awsaml-search" src="https://user-images.githubusercontent.com/2836432/43696583-43e0b454-9904-11e8-91c4-56d31dda8ffb.png">


GH-118

dgreene-r7 commented on src/containers/configure/RecentLogins.js:
```diff
@@ -15,28 +20,95 @@ const RecentLoginsHeader = styled.h4`
+const mapDispatchToProps = () => ({
+});
+
+export const RecentLogins = connect(mapStateToProps, mapDispatchToProps)(RecentLoginsComponent);
```
I don't think you need to connect the Redux state here.

dgreene-r7 commented on src/containers/configure/RecentLogins.js:
```diff
@@ -15,28 +20,95 @@ const RecentLoginsHeader = styled.h4`
+    const metadataUrls = this.props.metadataUrls;
+    const filterText = this.state.filterText;
+
+    if (filterText === undefined || filterText === '') {
```
This can be simplified as `if (!filterText) {`.

bturner-r7 commented:
Thanks @dgreene-r7 - I disconnected redux state and simplified the empty check.

tquetano-r7 commented on src/containers/configure/RecentLogins.js:
```diff
@@ -15,28 +19,89 @@ const RecentLoginsHeader = styled.h4`
+    });
+  };
+
+  filterMetadataUrls = () => {
```
theres no reason this needs to be an instance field ... this could easily be a pure standalone function if passed the metadata / filterText as parameters.

tquetano-r7 commented on src/containers/configure/RecentLogins.js:
```diff
@@ -15,28 +19,89 @@ const RecentLoginsHeader = styled.h4`
+                <Login
+                  key={url}
+                  pretty={name}
+                  profileId={i}
```
using the iteration index as an ID of any kind is very dangerous and de-optimizes react rerender cycles, u should consider using a consistent unique value.

bturner-r7 replied:
Great call-out.  This would have actually broken deleting a profile when URLs were filtered because the delete uses the iteration index.

Unfortunately, this is how the code was when I found it.  I've attempted to fix by starting to store a UUID with each profile and then referencing the UUID instead of the iteration index: c5470b1de09718acabf9f42eec0c4fde4ec2c9db

Thoughts?

tquetano-r7 commented on src/containers/configure/RecentLogins.js:
```diff
@@ -15,28 +19,89 @@ const RecentLoginsHeader = styled.h4`
+    filterText: '',
+  };
+
+  handleFilterInputChange = ({target: {value}}) => {
```
u may consider renaming `value` to take advantage of shorthand (and define the assignment contract in the initial line):

```javascript
handleFilterInputChange = ({target: {value: filterText}}) => this.setState({filterText});

bturner-r7 commented:
I think I've addressed the code review comments.  If @dgreene-r7 and @tquetano-r7 could take one more look I'd appreciate it.  Especially this commit: c5470b1de09718acabf9f42eec0c4fde4ec2c9db, which I added as a way to ensure deletes still work even when filtering URLs.
